File: binary-decimal.cpp
#include <iostream>
#include <cmath>
#include <string>

using namespace std;

string reverse_bin (string bin){
  string bin2 = "";
  for (int i=bin.size()-1; i>=0; i--){
    bin2+=bin[i];
  }
  return bin2;
}

string first (int num){
  string a = "";
  while (num>=1){
    if (num%2==0)
      a+='0';
    else
      a+='1';
    num/=2;
  }
  return reverse_bin (a);
}

long long int second (int num) {
  int sum = 0, i = 0;
  while (num>=1){
    sum+=pow(2, i)*(num%10);
    num/=10;
    i++;
  }
  return sum;
}

int main() {
  int ask, num;
  
  cout << "Which conversion do you want?\nWrite 0 for Decimal to Binary and 1 for Binary to Decimal: ";
  cin >> ask;
  
  cout << "Write the integer that you want to convert: ";
  cin >> num;
  
  if (ask==0)
    cout << "Your decimal number in binary is: " << first(num) << "\n";
  else
    cout << "Your binary number in decimal is: " << second(num) << "\n";
  
  return 0;
}

File: binary-decimal.py
def reverse_bin(x):
  bin = ''
  for i in range(len(x)-1, -1, -1):
    bin+=x[i]
  return bin

def first (x):
  bin = ''
  while x>=1:
    bin+=str(int(x%2))
    x/=2
  return reverse_bin(bin)

def second(x):
  new = reverse_bin(str(x))
  sum = 0
  for i in range (len(new)-1, -1, -1):
   sum+=(2**i)*int(new[i])
  return sum

ask = int(input("Which conversion do you want?\nWrite 0 for Decimal to Binary and 1 for Binary to Decimal: "))
num = int(input("Write the integer that you want to convert: "))

if ask == 0:
  print('Your decimal number in binary is: {}'.format(first(num)))
else:
  print('Your binary number in decimal is: {}'.format(second(num)))
